src/library/scala/collection/Iterable.scala

```diff
@@ -68,12 +68,12 @@ trait Iterable[+A] extends IterableOnce[A]
   protected[this] def stringPrefix: String = "Iterable"
 
   /** Converts this $coll to a string.
-    *
-    *  @return   a string representation of this collection. By default this
-    *            string consists of the `className` of this $coll, followed
-    *            by all elements separated by commas and enclosed in parentheses.
     */
-  override def toString = mkString(className + "(", ", ", ")")
+  override def toString: String  =
+    this match {
+      case _: StrictOptimizedIterableOps[_, _, _] => mkString(className + "(", ", ", ")")
+      case _                                      => className + "(<iterable>)"
+    }
```

Comment:
@szeiger Does this look ok?

Reply:
I feel like this is not really ok. According to the name of these `StrictOptimized______Ops`, they are meant to not change behaviour at all, just to provide performance optimizations for strict collections. But for instance, users who define their own collections may forget to inherit from `StrictOptimized________Ops`. Currently,  the only consequence is that they may not get the absolute best performance, but with these changes, they'll actually have broken `toString` methods.

Reply:
In `stringOf`, it's trying to accommodate custom iterables that have their own `toString`.